pynest/nest/lib/hl_api_connections.py

```diff
@@ -303,12 +303,21 @@ def Connect(pre, post, conn_spec=None, syn_spec=None, model=None):
                                     "scalar or a dictionary.")
                             else:
                                 syn_spec[key] = value
+                        elif rule == 'fixed_total_number':
+                            if ('N' in conn_spec
+                                    and value.shape[0] != conn_spec['N']):
```

Comment:
Can you use `fixed_total_number` without giving `N`? I don't think you need to check if `N` is in `conn_spec`.

Reply:
You're right in that `N` is required when using `fixed_total_number`, but I did it this way to make the error consistent. If we don't check if `N` is specified here, we would get a Python `KeyError`, while usually when using `fixed_total_number` without `N` we would get a `NESTErrors.DictError`.

Reply:
@hakonsbm Ok!